### PR TITLE
[PH][MIBM-126] update the /check-you-answers page and model

### PR DESCRIPTION
**Pull request by PaulHodgson**

### CODE REVIEW

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/PurchaseDetailsController.scala [hunk 1]
```diff
@@ -38,19 +38,19 @@ class PurchaseDetailsController @Inject()(
...
+    request.declarationJourney.goodsEntries.entries.headOption match {
       case Some(goodsEntry) =>
 
         getCurrencies().map { currencyPeriod =>
-          val preparedForm = goodsEntry.maybePriceOfGoods match {
-            case Some(priceOfGoods) => form.fill(priceOfGoods.toPurchaseDetailsInput)
+          val preparedForm = goodsEntry.maybePurchaseDetails match {
+            case Some(purchaseDetails) => form.fill(purchaseDetails.purchaseDetailsInput)
             case None => form
           }
```
Comment (PasGat) [inline app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/PurchaseDetailsController.scala:55]: few nested pattern matching, maybe extracting into smaller methods would help readability?

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/declaration/Declaration.scala [hunk 2]
```diff
@@ -103,28 +129,69 @@ object JourneyDetails {
...
+      _ <- maybeGoodsDestination
+
+      if maybeExciseOrRestrictedGoods.contains(false)
+      if maybeValueWeightOfGoodsExceedsThreshold.contains(false)
+      if (iAmACustomsAgent && maybeCustomsAgent.isDefined) || (!iAmACustomsAgent)
+      if journeyDetailsCompleteAndDeclarationRequired
+      if nameOfPersonCarryingTheGoods.firstName.trim.nonEmpty
+      if nameOfPersonCarryingTheGoods.lastName.trim.nonEmpty
+    } yield {
       Declaration(
```
Comment (PasGat) [inline app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/declaration/Declaration.scala:182]: few if statements in there :-). Would something like "exists" help? Or even better an Applicative

**Review by jordanrowe — APPROVED** [review]

**Review by PasGat — APPROVED** [review]

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/declaration/Declaration.scala (hunk 2 above)
Comment (PaulHodgson) [reply app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/declaration/Declaration.scala:182]: I had meant to remove some of those out as they shouldn't be handled this way. There's an update so now its just:

      if maybeExciseOrRestrictedGoods.contains(false)
      if maybeValueWeightOfGoodsExceedsThreshold.contains(false)
      if maybeCustomsAgent.isDefined || !iAmACustomsAgent
      if journeyDetailsCompleteAndDeclarationRequired

how's that look?

File: app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/PurchaseDetailsController.scala (hunk 1 above)
Comment (PaulHodgson) [reply app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/PurchaseDetailsController.scala:55]: I think these could all be re-factored to use fold:

val preparedForm = goodsEntry.maybePurchaseDetails fold(form) { purchaseDetails =>
            form.fill(purchaseDetails.purchaseDetailsInput)
}

there's a bunch that I could do on a follow-up pr?

**Pull request merged**